From 2a556c12f7af9f1fed61b59194a0dac855c49fdf Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Mon, 7 Apr 2014 20:02:08 +1000
Subject: [PATCH] Fix calculation error.

Should have no effect in practice as we never get close to the end of the ringbuffer while in read internal mode
---
 mythtv/libs/libmythtv/ringbuffer.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/ringbuffer.cpp b/mythtv/libs/libmythtv/ringbuffer.cpp
index 7f00478..b92921a 100644
--- a/mythtv/libs/libmythtv/ringbuffer.cpp
+++ b/mythtv/libs/libmythtv/ringbuffer.cpp
@@ -1475,7 +1475,7 @@ int RingBuffer::ReadPriv(void *buf, int count, bool peek)
     int rpos;
     if (rbrpos + readOffset > bufferSize)
     {
-        rpos = bufferSize - readOffset;
+        rpos = (rbrpos + readOffset) - bufferSize;
     }
     else
     {
-- 
1.7.10.2

